sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/OnTimerMethodSpecifier.java

```diff
@@ -0,0 +1,20 @@
+package org.apache.beam.sdk.transforms.reflect;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.transforms.DoFn;
+
+/**
+ * Used by {@link ByteBuddyDoFnInvokerFactory} to Dynamically generates
+ * {@link OnTimerInvoker} instances for invoking a particular
+ * {@link DoFn.TimerId} on a particular {@link DoFn}.
+ */
+
+@AutoValue
+abstract class OnTimerMethodSpecifier {
+    public abstract Class<? extends DoFn<?, ?>> fnClass();
+    public abstract String timerId();
+    public static OnTimerMethodSpecifier
+    create(Class<? extends DoFn<?, ?>> fnClass, String timerId){
```

Comment:
Since I am pretty new to the code base, I wanted to keep the risk of breaking something low. I do understand the value that the ```for``` or ```of``` construct brings. It makes the intention very clear. 
Thanks.